### ISSUE TITLE
Xamarin support - are current solutions "safe" until 2025?

### ISSUE THREAD

**Issue by milinkovicivana** [xamarin/Xamarin.Forms#15845]
Hi, 

We have some questions regarding Xamarin support, and hope that you can help us. 
Since we have implemented our Xamarin product (mbanking application) in many banks across Western Balkan, it's extremely important to us to know the final date with which our current solutions will be "safe", so that we can communicate it to our clients and plan the migration to MAUI.
We follow the latest information regarding this subject and we're familiar with announced support for XCode 15 and Android API 34 targeting, but just wanted to check if we can really "breathe a sigh of relief" until 2025?
We tried latest Visual Studio and managed to target Android 14, like described here https://learn.microsoft.com/en-us/xamarin/android/release-notes/13/13.2, will this be enough for us to be able to publish our app to Google store until next API target requirment (08/2025)?
We have the same question for iOS. We're already using XCode 15 for debugging and development, but can we count on you to provide support for new minor versions (example: 15.1) if Apple requires apps to be built with that version, so that we can publish to AppStore until next Apple requirment (05/2025)? 

Thanks in advance, every information help us.

**Comment by jfversluis** [comment]
Everything you need to know is described in the [official support policy](https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin).

We managed to add support for Xcode 15 and Android API 34, but only to make Xamarin build against these SDK versions. No new APIs that have been introduced in these SDK versions are supported in Xamarin, only on .NET MAUI. However, if you don't need those you should be fine.

If we assume that Apple will release Xcode 16 in Sept 2024, they usually have a 6 month period where they still allow apps and updates in the App Store built against Xcode 15. That means around May 2025. Google will have some similar policy.

However, the support from Microsoft on Xamarin products end on May 2024. You might very well still be able to build apps for another year and publish those to the stores, but if some breaking change happens, there will be no Xamarin update to fix this. Historically I don't think this ever happened, but theoretically it could.

So TL;DR: we keep things running until May 2024, after that there will be no updates for Xamarin. You should be able to publish to the stores well into 2025, but no guarantees after May 2024.

Of course it is all open-source so you might just fork it and build your own version if you really need it, but I would think your best option is to start looking into migrating to .NET MAUI sooner rather than later.
